shell/types/resources/fleet.d.ts

```diff
@@ -0,0 +1,24 @@
+export interface ResourceKey {
```

Comment:
these should all be prefixed with something more resource specific. not sure on specifics, but something like

BundleDeploymentResourceKey
BundleDeploymentModifiedStatus
BundleDeploymentNonReadyBundle

This helps a LOT as there's so much crossover of terms in the UI and all the products it serves